Sofle Shield: Add underglow support and increase encoder sensitivity

infused-kim opened the pull request:
This PR adds underglow support to the sofle keyboard. I used the kyria shield as a reference.

It's been tested on a sofle choc, but the pins are the same on the sofle RGB and there should be no issues.


## Board/Shield Check-list
 - [x] This board/shield is tested working on real hardware
 - [x] Definitions follow the general style of other shields/boards upstream ([Reference](https://zmk.dev/docs/development/new-shield))
 - [x] `.zmk.yml` metadata file added
 - [x] Proper Copyright + License headers added to applicable files (Generally, we stick to "The ZMK Contributors" for copyrights to help avoid churn when files get edited)
 - [x] General consistent formatting of DeviceTree files
 - [x] Keymaps do not use deprecated key defines (Check using the [upgrader tool](https://zmk.dev/docs/codes/keymap-upgrader))
 - [x] `&pro_micro` used in favor of `&pro_micro_d/a` if applicable
 - [x] If split, no name added for the right/peripheral half
 - [x] Kconfig.defconfig file correctly wraps *all* configuration in conditional on the shield symbol
 - [x] `.conf` file has optional extra features commented out
 - [x] Keyboard/PCB is part of a shipped group buy or is generally available in stock to purchase (OSH/personal projects without general availability should create a zmk-config repo instead)

caksoylar commented on app/boards/shields/sofle/sofle.dtsi:
```diff
@@ -49,7 +49,7 @@ RC(3,0) RC(3,1) RC(3,2) RC(3,3) RC(3,4) RC(3,5) RC(4,5) RC(4,6) RC(3,6) RC(3,7)
         a-gpios = <&pro_micro 21 (GPIO_ACTIVE_HIGH | GPIO_PULL_UP)>;
         b-gpios = <&pro_micro 20 (GPIO_ACTIVE_HIGH | GPIO_PULL_UP)>;
-        resolution = <4>;
+        resolution = <2>;
```
I believe this depends on the encoder hardware itself and not the shield. IMO it should be set the same for all shields (to the most common value if possible) then the user should override it in their config if their encoder is different.

infused-kim replied:
Ah, I didn't realize that. I just noticed my encoder was skipping steps.

No problem, I can remove that commit if that helps getting the changes merged.

infused-kim replied:
I have removed this now.

infused-kim commented:
I have removed the encoder sensitivity commit as per @caksoylar's suggestion and updated the underglow settings for zephyr 3.

Is there anything else that is necessary to be done to get this merged?

Nicell commented:
@infused-kim It looks like this branch is failing to build ATM: https://github.com/infused-kim/zmk/runs/6226559733
Is this just because it's so far behind main it doesn't have the updates requiring color mapping?

Cringely commented:
Would love for this to be merged, thanks for the effort @infused-kim!

autoferrit commented:
I recently acquired this board. I would love to see this merged. I am new to C and ZMK but would like to know if there is anything I can do to help get this merged in.

SethMilliken commented:
I've run across a few more instances on various Discord servers of people having issues enabling underglow on their Sofle. Getting these changes landed seems like very low-hanging fruit with no downside. I would be happy to update this PR with the 3.2 `pinctrl` changes if that helps.

SethMilliken commented:
These changes ([updated](https://github.com/zmkfirmware/zmk/pull/1854) for Zephyr 3.2) have now been merged! This PR can be closed now.
